### PR TITLE
Make inspect-rewrite wait for start

### LINKED ISSUE #1549: weaveproxy --rewrite-inspect race at container startup
If you call `docker inspect` on a container right after it starts, you will get the docker IP address until `weave attach` has finished setting up the weave IP address.

This is on both 1.1 and on master.

Weaveproxy could note which containers are still in initialization and hold up any `inspect` requests until initialization has finished.

### PR DESCRIPTION
**Pull request by bboreham**
Fixes #1549

This seems difficult to write a smoke-test for, as `docker start` won't return until the start is finished; the issue shows up if you react to the 'start' event and do an inspect straight away.

### CODE REVIEW

**Comment by rade** [comment]
provided this doesn't get more complex, I think it is safe to stick this in 1.2.1.

@bboreham is there anything left to do here?

**Comment by bboreham** [comment]
@rade nothing left from me.

**Comment by rade** [comment]
@paulbellamy pls review. If you are happy with it then lets rebase on 1.2.

File: proxy/inspect_container_interceptor.go [hunk 1]
```diff
@@ -22,19 +22,20 @@ func (i *inspectContainerInterceptor) InterceptResponse(r *http.Response) error
 		return err
 	}
 
-	if err := updateContainerNetworkSettings(container); err != nil {
```
Comment (paulbellamy) [inline proxy/inspect_container_interceptor.go]: I'd prefer this as a method on proxy, than taking proxy as an arg.

Comment (rade) [inline proxy/inspect_container_interceptor.go]: agreed. especially since it's already shared by multiple interceptors.

File: proxy/proxy.go [hunk 2]
```diff
@@ -357,6 +357,23 @@ func (proxy *Proxy) waitForStart(r *http.Request) {
 	}
 }
 
+// If some other operation is waiting for a container to start, join in the wait
+func (proxy *Proxy) waitForStartByIdent(ident string) {
```
Comment (paulbellamy) [inline proxy/proxy.go]: Should probably return out an error, like `waitForStart`?

File: proxy/inspect_container_interceptor.go (hunk 1 above)
Comment (bboreham) [inline proxy/inspect_container_interceptor.go]: also move the implementation to proxy.go?

Comment (paulbellamy) [inline proxy/inspect_container_interceptor.go]: please

**Comment by bboreham** [comment]
Updated in line with review comments and rebased on 1.2.  Will create a new PR to merge into 1.2.

**Pull request closed without merging**